### ISSUE TITLE
[Question] question title whether my ppo implementation for sumo-rl env is converged?

### ISSUE THREAD

**Issue by Gavin-Tao** [DLR-RM/stable-baselines3#2240]
### ❓ Question

<img width="1845" height="648" alt="Image" src="https://github.com/user-attachments/assets/07db659f-2305-4f42-9c14-11c5644f41c0" />

<img width="1839" height="666" alt="Image" src="https://github.com/user-attachments/assets/d89bec97-f5c3-442f-bf9f-aa376ef87507" />

Hi, I'm training a PPO agent for single-intersection traffic signal control using SUMO-RL. The environment has discrete actions (4 phases) and uses a priority-pressure based reward function that accumulates vehicle queue values each step, resulting in large reward magnitudes.
Hyperparameters:

n_steps: 2000, n_epochs: 10, batch_size: 64
ent_coef: 0.15, vf_coef: 0.5, lr: 3e-4
gamma: 0.99, gae_lambda: 0.95, clip_range: 0.2
Episode length: 1000s / delta_time: 5s = 200 steps per episode
No VecNormalize applied

Observations from TensorBoard after ~1.1M steps:

value_loss: Still decreasing (~738k, not plateauing)
policy_gradient_loss: Stable around -0.0024
explained_variance: ≈ 5.96e-8 (effectively zero throughout training)
entropy_loss: Plateaued around -1.33, which is close to the maximum entropy for 4 discrete actions (-log(4) ≈ -1.386), suggesting the policy is stuck near uniform sampling — likely because ent_coef is too large
clip_fraction: Near zero with occasional spikes

My questions:

Is explained_variance ≈ 0 expected when reward scale is very large (value_loss in the hundreds of thousands)? Does this mean the advantage estimates are essentially noise?
Is wrapping with VecNormalize(norm_reward=True) the correct fix here, or should I scale the reward directly in the environment?
Given that entropy_loss has plateaued near maximum entropy, should I reduce ent_coef significantly (e.g. to 0.001)?
Is there any meaningful policy learning happening when explained_variance ≈ 0, or is the policy gradient update direction essentially random?

Thanks!

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [x] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [x] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> Is wrapping with VecNormalize(norm_reward=True) the correct fix here, or should I scale the reward directly in the environment?

Both are valid options.

**Comment by Gavin-Tao** [comment]
Please be aware that this email box is not available now. If you have any contact with me, please send your emails to my current email box ***@***.***).

**Comment by Gavin-Tao** [comment]
Please be aware that this email box is not available now. If you have any contact with me, please send your emails to my current email box ***@***.***).
